src/Core/System/Consent/ConsentRepository.php

```diff
@@ -0,0 +1,125 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\System\Consent;
+
+use Doctrine\DBAL\Connection;
+use Shopware\Core\Defaults;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Uuid\Uuid;
+use Shopware\Core\System\Consent\DTO\ConsentStateLogRecord;
+use Shopware\Core\System\Consent\DTO\ConsentStateRecord;
+
+/**
+ * @internal
+ *
+ * @codeCoverageIgnore integration tested with \Shopware\Tests\Integration\Core\System\Consent\ConsentRepositoryTest
+ */
+#[Package('data-services')]
+class ConsentRepository
+{
+    public function __construct(private readonly Connection $connection)
+    {
+    }
+
+    /**
+     * @return list<ConsentStateRecord>
+     */
+    public function fetchAllConsentStates(): array
+    {
+        $result = $this->connection->fetchAllAssociative(
+            'SELECT name, identifier, state, consent_state.actor_id FROM consent_state'
+        );
+
+        return array_map(
+            fn (array $row) => new ConsentStateRecord(
+                $row['name'],
+                $row['identifier'] ? Uuid::fromBytesToHex($row['identifier']) : null,
+                ConsentStatus::from($row['state']),
+                Uuid::fromBytesToHex($row['actor_id'])
+            ),
+            $result
+        );
+    }
+
+    public function updateConsentState(
+        ConsentDefinition $consent,
+        ?string $identifier,
```

Comment:
i can pass the identifier independend of the consent definition, which makes it brittle, as when i update a consent that has global scope, but add a scope identifier here suddenly it is treated as it would be scoped